crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -390,6 +390,19 @@ where
             return Ok(OnForkChoiceUpdated::invalid_state())
         }
 
+        // TODO: make this less complex - how should we track invalid parents for a given fcu
+        // state?
+        let lowest_buffered_ancestor_fcu = self
+            .blockchain
+            .lowest_buffered_ancestor(state.head_block_hash)
+            .map(|ancestor| ancestor.parent_hash)
+            .unwrap_or_else(|| state.head_block_hash);
+
+        if let Some(status) = self.check_invalid_ancestor(lowest_buffered_ancestor_fcu) {
+            // TODO: confusing - valid is returned here but the status is invalid
+            return Ok(OnForkChoiceUpdated::valid(status))
```

Comment:
right, should rename this function to `from_status` or something

Reply:
created `with_invalid`, similar to the valid one